lib/charms/observability_libs/v0/kubernetes_service_patch.py

```diff
@@ -221,6 +251,10 @@ def is_patched(self) -> bool:
         fetched_ports = [(p.port, p.targetPort) for p in service.spec.ports]  # type: ignore[attr-defined]  # noqa: E501
         return expected_ports == fetched_ports
 
+    def _on_remove(self, event):
```

Comment:
Curious why you added this? Assuming the juju labels persist, juju should automatically remove the service when the charm is removed?

Reply:
Labels persist indeed but juju doesn't automatically remove the service on remove. I believe that it's because the 'app.kubernetes.io/name' label points to the new service name and that juju looks for the application that it's trying to remove.